tests/traverse.js

```diff
@@ -0,0 +1,35 @@
+import esquery from '../esquery.js';
+import conditional from './fixtures/conditional.js';
+
+describe('traverse', function () {
+    it('iterates matches', function () {
+        const matches = [];
+        const parents = [];
+        const ancestries = [];
+        const selector = esquery.parse(':matches(IfStatement)');
+        esquery.traverse(conditional, selector, (match, parent, ancestry) => {
+            parents.push(parent);
+            matches.push(match);
+            ancestries.push(ancestry);
+        });
+        assert.includeMembers(matches, [
```

Comment:
I'd like to see assertions that there's also nothing else in here. You can do this by asserting equal `length`s.

Reply:
In fact, we should probably guarantee an order. Even if we don't want them to, there's really no way of stopping someone from relying on the traversal order once this is exposed. So just assert array equality.

Reply:
Gotcha. I've switched to `deepEqual` which I've confirmed does not allow subsets or supersets, nor members in different order, so that should guarantee a matching length and order (as well as content).